staging/src/k8s.io/apiserver/pkg/admission/plugin/resourcequota/resource_access.go

```diff
@@ -119,7 +119,7 @@ func (e *quotaAccessor) GetQuotas(namespace string) ([]corev1.ResourceQuota, err
 			// use singleflight.Group to avoid flooding the apiserver with repeated
 			// requests. See #22422 for details.
 			lruItemObj, err, _ = e.group.Do(namespace, func() (interface{}, error) {
-				liveList, err := e.client.CoreV1().ResourceQuotas(namespace).List(context.TODO(), metav1.ListOptions{})
+				liveList, err := e.client.CoreV1().ResourceQuotas(namespace).List(context.TODO(), metav1.ListOptions{ResourceVersion: "0"})
```

Comment:
This change will break this scenario:

1. global resourcequota configuration defaulting to allow no usage of a particular resource (e.g. system-critical pods or some storage class or something)
2. namespace initialization process that rapidly in order creates a) ResourceQuota object allowing use of the globally-restricted resource then b) an object that uses that resource (like a privileged pod or PersistentVolumeClaim or something)

If we're going to make the "live" lookup work from cache, we should just delete the entire `if len(items) == 0 {` block because there's no real difference between the client-held informer-fed cache and the server-held informer-fed cache used with `ResourceVersion=0`.

Reply:
From https://github.com/kubernetes/design-proposals-archive/blob/main/resource-management/admission_control_resource_quota.md:

> To optimize system performance, it is encouraged that all resource quotas are tracked on the same ResourceQuota document in a Namespace. As a result, it is encouraged to impose a cap on the total number of individual quotas that are tracked in the Namespace to 1 in the ResourceQuota document.

It is also mentioned in [the documentation](https://kubernetes.io/docs/concepts/policy/resource-quotas/). However, based on the implementation it seems it is not "optimization", but a hard requirement for quota to behave in consistent matter.

Reply:
> we need to double check if there is any case that depends on this lister not being initialized.

I think Jordan described this above with his 2.  It happens whenever a person does a `kubectl create -f dir` where dir contains namespace, followed by resourcequota, followed by limitrange, followed by PVC.  If the cache is latent, the PVC created would not have the expected LimitRange applied.

Reply:
actually... digging deeper, quota admission triggers both for creation of new namespaces and creation of resourcequota objects, and this fallback to a live lookup gets *cached* for 30 seconds

That means if a user rapidly creates a namespace + quota + other stuff, the namespace create will trigger the live lookup and cache a list of zero resource quota objects for 30 seconds.

It also means that if a user in a namespace with no quotas where nothing was recently created rapidly creates a quota + other stuff, the quota create will trigger the live lookup and cache a list of zero resource quota objects for 30 seconds.

This effectively means that the lister was the only useful part of this.

I think we should drop the `if len(items) == 0 {` `liveLookupCache` bit completely for quota.

Reply:
>   I think we should drop the if len(items) == 0 { liveLookupCache bit completely for quota.

No issue. My example above for LimitRange concerns me since we can't mutate the PVC/pod. Checking code, defaults are only for Pods it appears.

Reply:
Yeah, let's deal with resourcequota and limitranger separately.